File: backend/WatchNow/src/main/java/com/intiformation/WatchNow/service/OeuvreService.java
package com.intiformation.WatchNow.service;

import java.util.List;

import com.intiformation.WatchNow.model.Oeuvre;

public interface OeuvreService {

	Oeuvre getById(String id);
	List<Oeuvre> getByTitre(String titre);
	List<Oeuvre> getByTitre(String titre, Integer nbResults);
	List<Oeuvre> getMostPopularByType(String type);
	List<Oeuvre> getMostPopularByType(String type, Integer nbResults);
	List<Oeuvre> getMostPopularByTypeAndGenre(String type, String genre);
	List<Oeuvre> getMostPopularByTypeAndGenre(String type, String genre, Integer nbResults);
	List<Oeuvre> getMoreLikeThis(String id);
	List<Oeuvre> getMoreLikeThis(String id, Integer nbResults);
	List<Oeuvre> getByKeyword(String keyword);
	List<Oeuvre> getByKeyword(String keyword, Integer nbResults);
	List<Oeuvre> getComingSoonByType(String type);
	List<Oeuvre> getComingSoonByType(String type, Integer nbResults);
	List<Oeuvre> getByTitreAndType(String titre, String type);
	List<Oeuvre> getByTitreAndType(String titre, String type, Integer nbResults);
	List<Oeuvre> getByTitreAndAnnee(String titre, int annee);
	List<Oeuvre> getByTitreAndAnnee(String titre, int annee, Integer nbResults);
	List<Oeuvre> getByTitreAndTypeAndAnnee(String titre, String type, int annee);
	List<Oeuvre> getByTitreAndTypeAndAnnee(String titre, String type, int annee, Integer nbResults);
}

File: frontend/WatchNow/src/app/service/oeuvre.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Oeuvre } from '../model/oeuvre.model';
import { FilmService } from './film.service';
import { SerieService } from './serie.service';

@Injectable({
  providedIn: 'root'
})
export class OeuvreService {

  constructor(private httpCLient: HttpClient, private filmService:FilmService, private serieService:SerieService) { }

  getByKeyword(keyword:string): Observable<Oeuvre[]> {
    return this.httpCLient.get<Oeuvre[]>(environment.url + "oeuvre/index/keyword/" + keyword)
  }

  getByMostPopularByType(type:string) : Observable<any[]> {
    if(type == "film") {
      return this.filmService.getMostPopularFilm();
    }else {
      return this.serieService.getMostPopularSerie();
    }
  }
  
  getByTitre(titre:string): Observable<Oeuvre[]> {
    return this.httpCLient.get<Oeuvre[]>(environment.url +"oeuvre/oeuvre/titre/" + titre + "?nbResults=8")
  }

  getByTitreAndtype(type:string, titre:string): Observable<Oeuvre[]> {
    return this.httpCLient.get<Oeuvre[]>(environment.url +"oeuvre/oeuvre/titre/" + titre + "/type/" + type + "?nbResults=8")
  }

  getByTitreAndAnnee(titre:string, annee:string): Observable<Oeuvre[]> {
    return this.httpCLient.get<Oeuvre[]>(environment.url +"oeuvre/oeuvre/titre/" + titre + "/annee/" + annee + "?nbResults=8")
  }

  getByTitreAndTypeAndAnnee(titre:string, type:string, annee:string): Observable<Oeuvre[]> {
    return this.httpCLient.get<Oeuvre[]>(environment.url +"oeuvre/oeuvre/titre/" + titre + "/type/" + type +"/annee/" + annee + "?nbResults=8")
  }


}
